ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -5,21 +5,105 @@
 
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
 
 export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
   // =services
 
   @service store;
+  @service confirm;
+  @service ipc;
+  @service router;
+  @service session;
+
+  queryParams = {
+    isConnecting: {
+      refreshModel: true,
+    },
+  };
 
   // =methods
 
   /**
    * Load a target
    * @param {object} params
    * @param {string} params.target_id
+   * @param {bool} params.isConnecting
    * @return {TargetModel}
    */
-  model({ target_id }) {
-    return this.store.findRecord('target', target_id, { reload: true });
+  async model({ target_id, isConnecting }) {
+    const target = await this.store.findRecord('target', target_id, {
+      reload: true,
+    });
+
+    if (isConnecting) {
+      this.preConnect(target);
+    }
+    return target;
+  }
+
+  /**
+   * Determine if connection workflow based on target attributes.
+   * @param {TargetModel} model
+   */
+  @action
+  @loading
+  async preConnect(target) {
+    if (target.address) {
+      this.connect(target);
+    }
+  }
+
+  /**
+   * Establish a session to current target.
+   * @param {TargetModel} model
+   * @param {HostModel} host
+   */
+  @action
+  @loading
+  async connect(model, host) {
+    // TODO: Connect: move this logic into the target model
```

Comment:
Is this TODO comment still valid?

Reply:
That is a good question that I planned to raise to the pod/team. This is from when the connect method was originally written. Not sure if it needs to live on the target model or not. Happy to discuss with the pod tomorrow.

Reply:
Awesome!! Ideed, good for discussion 😉